src/Gatekeeper.php

```diff
@@ -143,6 +142,10 @@ function () {
                 );
 
                 if ($this->isGuest() && !$this->isAuthenticated() && !$this->isGatekeeperRequest()) {
+                    $cookie = new Cookie(['name' => 'gatekeeper_referer']);
+                    $cookie->value = Craft::$app->getRequest()->getUrl();
+                    $cookie->expire = time() + 20;
```

Comment:
Should this cookie live a bit longer in case you type the wrong password or need to look it up somewhere before logging in?

Reply:
Oh, sorry @boscho87. I see you already mentioned it in the PR. I agree that it should last longer. Thoughts @tomdiggle?

Reply:
I've increased the expire time to 30 seconds. Hopefully that'll be enough time if the password is entered incorrectly.